src/mds/Server.cc

```diff
@@ -6832,7 +6835,7 @@ void Server::_link_rollback_finish(MutationRef& mut, MDRequestRef& mdr,
   if (mdr)
     mdcache->request_finish(mdr);
 
-  mdcache->finish_rollback(mut->reqid);
+  mdcache->finish_rollback(mdr);
```

Comment:
mdr is null if mds is in resolve state. MDCache::finish_rollback should accept both (metareqid_t reqid,  MDRequestRef& mdr)